src/binaryen-c.cpp

```diff
@@ -3578,6 +3598,82 @@ void BinaryenSetMemory(BinaryenModuleRef module,
   }
 }
 
+// Memory segments
+
+uint32_t BinaryenGetNumMemorySegments(BinaryenModuleRef module) {
+  if (tracing) {
+    std::cout << "  BinaryenGetNumMemorySegments(the_module);\n";
+  }
+
+  auto* wasm = (Module*)module;
+  return wasm->memory.segments.size();
+}
+int64_t BinaryenGetMemorySegmentByteOffset(BinaryenModuleRef module, BinaryenIndex id) {
+  if (tracing) {
+    std::cout << "  BinaryenGetMemorySegmentByteOffset(the_module, " << id
+              << ");\n";
+  }
+
+  auto* wasm = (Module*)module;
+  if (wasm->memory.segments.size() <= id) {
+    Fatal() << "invalid segment id.";
+  }
+  Memory::Segment segment = wasm->memory.segments[id];
+
+  int64_t ret;
+  auto globalOffset = [&](const Expression* const & expr, int64_t & result) -> bool {
+    if (auto* c = expr->template dynCast<Const>()) {
+      result = c->value.getInteger();
+      return true;
+    }
+    return false;
+  };
+
+  if ( globalOffset(segment.offset, ret) ) {
```

Comment:
Can you run clang-format on this file? This should catch the extra whitespace in the parens, and possibly other things I haven't caught.